internal/cel/cel.go

```diff
@@ -0,0 +1,129 @@
+package cel
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/ext"
+	"k8s.io/apiserver/pkg/cel/library"
+)
+
+type Compiler struct {
+	env *cel.Env
+}
+
+// variables is a placeholder type for the variables object.
+// This is needed in the parse/check phase, where the variables
+// are added sequentially to the CEL environment.
+// It's not possible to use a cel.DynType, because the compiler
+// will not complain about undeclared variables.
+type variables struct{}
+
+func NewCompiler() (*Compiler, error) {
+	env, err := cel.NewEnv(
+		// Kubernetes 1.29 options
+		cel.HomogeneousAggregateLiterals(),
+		cel.EagerlyValidateDeclarations(true),
+		cel.DefaultUTCTimeZone(true),
+		cel.CrossTypeNumericComparisons(true),
+		cel.OptionalTypes(),
+		cel.ASTValidators(
+			cel.ValidateDurationLiterals(),
+			cel.ValidateTimestampLiterals(),
+			cel.ValidateRegexLiterals(),
+			cel.ValidateHomogeneousAggregateLiterals(),
+		),
+
+		// Kubernetes 1.29 libraries and extensions
+		ext.Sets(),
+		ext.Strings(ext.StringsVersion(2)),
+		library.URLs(),
+		library.Regex(),
+		library.Lists(),
+		// TODO: introduce these libraries
```

Comment:
Maybe add github issues about that into the repo

Reply:
tracked here: https://github.com/kubewarden/cel-policy/issues/6